source/slang/slang-emit-glsl.cpp

```diff
@@ -310,7 +334,16 @@ void GLSLSourceEmitter::_emitGLSLParameterGroup(IRGlobalParam* varDecl, IRUnifor
     or IRGLSLShaderStorageBufferType which is read write.
     */
 
-    _emitGLSLLayoutQualifier(LayoutResourceKind::DescriptorTableSlot, &containerChain);
+    {
+        // TODO(JS):
+        // The *assumption* here is that we only need to detect the LayoutResourceKind as used on the initial containerChain.varLayout
+        // rather than search up the chain. 
+        //
+        // Perhaps there is still an issue here, because perhaps it's possible (?) to have a mixture of ConstantBuffer/DescriptorSlot
+        // and in that case the usedKind would just restrict to one and so produce the wrong answer.
```

Comment:
I believe that it would *not* be possible to have a mix of multiple `LayoutResourceKind`s at this point, since a single leaf shader variable of resource type should only have a single `LayoutResourceKind`, and type legalization should have split up any types that would violate that rule (since they'd be illegal in GLSL).

(It might be good to validate this belief by looking at the state inside the debugger, before making any changes based on my assumption...)